### ISSUE TITLE
Possible crash "Floating point exception" when word_timestamps=True

### ISSUE THREAD

**Issue by Jeronymous** [guillaumekln/faster-whisper#339]
The following code systematically fails with the enclosed wave audio (which is 33.84 sec long), after printing:
```
s.id=1 | s.seek=3000 | s.start=4.06 | s.end=12.32
s.id=2 | s.seek=3000 | s.start=12.72 | s.end=19.32
s.id=3 | s.seek=3000 | s.start=19.88 | s.end=25.16
s.id=4 | s.seek=3000 | s.start=25.6 | s.end=32.2
Floating point exception (core dumped)
```
I tested both version `0.6.0` from pip and latest master ([3b4a6aa](https://github.com/guillaumekln/faster-whisper/tree/3b4a6aa1c22d293ddde9f08bdd31fc842086a6ea)).

The same code does NOT fail with `word_timestamps=False`, and prints:
```
s.id=1 | s.seek=3000 | s.start=3.82 | s.end=12.6
s.id=2 | s.seek=3000 | s.start=12.6 | s.end=19.64
s.id=3 | s.seek=3000 | s.start=19.64 | s.end=25.48
s.id=4 | s.seek=3000 | s.start=25.48 | s.end=32.68
s.id=5 | s.seek=3001 | s.start=33.82 | s.end=35.82
s.id=6 | s.seek=3001 | s.start=35.82 | s.end=37.82
```
Note the high value of seek>3000 and the fact that timestamps go beyond the length of the audio.
So I guess it is due to a corner case in the alignment procedure when things happen too close to the end of a segment.

* wave audio: [audio.zip](https://github.com/guillaumekln/faster-whisper/files/11970051/audio.zip)
* code to reproduce:
```
import faster_whisper

audio = "audio.wav" # enclosed here as a zip

model = faster_whisper.WhisperModel("small", device="cpu", compute_type="int8")

segments, info = model.transcribe(audio,
    language="ru",
    word_timestamps=True,
    max_initial_timestamp=10000.0,
    vad_filter=True,
    beam_size=5, best_of=5, temperature= (0.0, 0.2, 0.4, 0.6, 0.8, 1.0),
    condition_on_previous_text=False,
    no_speech_threshold=1, compression_ratio_threshold=2.4
)

for s in segments:
    print(f"{s.id=} | {s.seek=} | {s.start=} | {s.end=}")
```

**Comment by Jeronymous** [comment]
The crash happens in the `align` method of the model, when `num_frames=1`
https://github.com/guillaumekln/faster-whisper/blob/3b4a6aa1c22d293ddde9f08bdd31fc842086a6ea/faster_whisper/transcribe.py#L824

**Comment by guillaumekln** [comment]
This error should be fixed in `ctranslate2>=3.12`. Can you try updating this module?

**Comment by Jeronymous** [comment]
Thanks a lot @guillaumekln 
Indeed updating `ctranslate2` solved that.
